src/pivot-table/hooks/__tests__/use-selections-model.test.ts

```diff
@@ -53,41 +56,46 @@ describe("useSelectionsModel", () => {
 
     // eslint-disable-next-line @typescript-eslint/unbound-method
     await waitFor(() => expect(selections.begin).toHaveBeenCalled());
+    await waitFor(() => expect(selections.on).toHaveBeenCalled());
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledWith({ currentPage: 0 }));
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 1)).toBeTruthy());
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 2)).toBeTruthy());
   });
 
   test("should select cell and not call begin selection when already active", async () => {
     selections.isActive = () => true;
-    const { result } = renderHook(() => useSelectionsModel(selections));
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
 
     await act(async () => {
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
     });
 
     // eslint-disable-next-line @typescript-eslint/unbound-method
     await waitFor(() => expect(selections.begin).toHaveBeenCalledTimes(0));
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledWith({ currentPage: 0 }));
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 1)).toBeTruthy());
   });
 
   test("should de-select cell", async () => {
-    const { result } = renderHook(() => useSelectionsModel(selections));
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
 
     await act(async () => {
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
     });
 
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 1)).toBeTruthy());
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledWith({ currentPage: 0 }));
 
     await act(async () => {
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
     });
 
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledWith({ currentPage: 0 }));
```

Comment:
are the selections being confirmed here? Otherwise upadatePageInfo sholudn't be called, right?

Reply:
You are right, selections are not confirmed here. In fact, we cannot mock the confirm behaviour in all of these test cases here(or at least i have no idea). And to answer for the question of why the test case is still working currently: this happens because of event registration in useEffect where we register all 4 events. By running the hook, this effect triggers and seems like those functions running once in that step! 

So for now it's better to only check it with something like `toHavebeenCalledTimes(1)` to make sure it's registered correctly.